VersionControlUI/src/DiffManager.cpp

```diff
@@ -766,13 +856,30 @@ void DiffManager::createOverlaysForChanges(Visualization::ViewItem* diffViewItem
 			{
 				auto overlay = new Visualization::ArrowOverlay{oldNodeItem, newNodeItem,
 							Visualization::ArrowOverlay::itemStyles().get("modify_arrow")};
-				diffViewItem->addOverlay(overlay, modifyArrowLayer);
+				if (isNameChangeRelated)
+					diffViewItem->addOverlay(overlay, NAME_CHANGE_ARROW_OVERLAY_NAME);
+				else
+					diffViewItem->addOverlay(overlay, modifyArrowLayer);
 			}
 		}
 
 	}
 
-	scaleItems(allItemsToScale, diffViewItem);
+	if (!itemsToScale.isEmpty())
+	{
+		scaleItems(itemsToScale, diffViewItem);
+
+		nameChangesScaledByAncestor_ += findAllItemsWithAncestorsIn(nameChangeItemsToScale, itemsToScale);
+
+	}
+
+	if (!nameChangeItemsToScale.isEmpty())
+	{
+		nameChangeItemsToScale.subtract(nameChangesScaledByAncestor_);
+		scaleItems(nameChangeItemsToScale, diffViewItem, true);
+	}
+
+	diffViewItem->setUpdateNeeded(Visualization::Item::StandardUpdate);
```

Comment:
Wow, this name change stuff resulted in a ton more changes than I expected.

I'm thinking, it might make sense to move all name change related things to a separate class. Let's talk about this in our next meeting. Think about it in the mean time.